### PR TITLE
Add API for accessing tracking confidence

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This PR improves the way we're parsing our location information and uses the `locationFlags` returned to determine how accurate our tracking is.

If `locationFlags` is 0 our confidence will be set to `TRACKING_CONFIDENCE_NONE (0)`.
If our valid bits are set but our tracking bits are not, confidence will be set to `TRACKING_CONFIDENCE_LOW (1)`
If both our set our confidence will be set to `TRACKING_CONFIDENCE_HIGH (2)`

These bits are tracked for both rotation and orientation, if both are valid the both need to be tracked for the confidence to be considered high.

You have access to the confidence directly on the `OpenXRPose` node, for ARVRController nodes we do not expose this so I've added a convenience method to `OpenXRConfig` called `get_tracking_confidence`. It takes the controller id as a parameter (so 1 for left hand, 2 for right hand, same as the ARVRControllers).

### CODE REVIEW

File: demo/addons/godot-openxr/CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Changes to the Godot OpenXR asset
 1.1.2
```
Comment (m4gr3d) [inline demo/addons/godot-openxr/CHANGES.md:4]: This should be bumped to `1.2.0`.

File: src/openxr/OpenXRApi.h [hunk 2]
```diff
@@ -52,6 +52,12 @@
 // forward declare this
 class OpenXRApi;
 
+enum TrackingConfidence {
+	TRACKING_CONFIDENCE_NO,
```
Comment (m4gr3d) [inline src/openxr/OpenXRApi.h:56]: This should be `TRACKING_CONFIDENCE_NONE` to match the version in the Godot source code.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: demo/addons/godot-openxr/CHANGES.md (hunk 1 above)
Comment (BastiaanOlij) [reply demo/addons/godot-openxr/CHANGES.md]: Done

File: src/openxr/OpenXRApi.h (hunk 2 above)
Comment (BastiaanOlij) [reply src/openxr/OpenXRApi.h]: Good point, done!

**Review by m4gr3d — APPROVED** [review]

**Pull request merged**